Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -53,40 +57,52 @@ void FiniteDifferenceSolver::ApplySilverMuellerBoundary (
     // Ensure that we are using the cells the domain
     domain_box.enclosedCells();
 
-#ifdef WARPX_DIM_RZ
+#if defined(WARPX_DIM_RZ) || defined(WARPX_DIM_RCYLINDER) || defined(WARPX_DIM_RSPHERE)
     // Calculate relevant coefficients
     amrex::Real const cdt = PhysConst::c*dt;
     amrex::Real const cdt_over_dr = cdt*m_h_stencil_coefs_r[0];
     amrex::Real const coef1_r = (1._rt - cdt_over_dr)/(1._rt + cdt_over_dr);
     amrex::Real const coef2_r = 2._rt*cdt_over_dr/(1._rt + cdt_over_dr) / PhysConst::c;
     amrex::Real const coef3_r = cdt/(1._rt + cdt_over_dr) / PhysConst::c;
+#if defined(WARPX_DIM_RZ)
     amrex::Real const cdt_over_dz = cdt*m_h_stencil_coefs_z[0];
     amrex::Real const coef1_z = (1._rt - cdt_over_dz)/(1._rt + cdt_over_dz);
     amrex::Real const coef2_z = 2._rt*cdt_over_dz/(1._rt + cdt_over_dz) / PhysConst::c;
+#endif
 
+#if !defined(WARPX_DIM_RSPHERE)
     // Extract stencil coefficients
     Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
     auto const n_coefs_z = static_cast<int>(m_h_stencil_coefs_z.size());
+#endif
 
     // Extract cylindrical specific parameters
     Real const dr = m_dr;
-    int const nmodes = m_nmodes;
     Real const rmin = m_rmin;
+#if !defined(WARPX_DIM_RSPHERE)
```

Comment:
```suggestion
#if !defined(WARPX_DIM_RSPHERE)
```

Reply:
This was changed to ```#if defined(WARPX_DIM_RZ)``` and changes were more below to only include the modes when RZ.